### PR TITLE
Part seven

### PR DESCRIPTION
**Pull request by sheplu**
Hello !

Un peu (beaucoup) de retard, mais voila la partie 7

J'attends vos retours

### CODE REVIEW

File: docs/tutorial/part-seven/index.md [hunk 1]
```diff
@@ -1,61 +1,54 @@
 ---
-title: Programmatically create pages from data
+title: Créer des pages dynamiquement depuis des données
 typora-copy-images-to: ./
 disableTableOfContents: true
 ---
 
-> This tutorial is part of a series about Gatsby’s data layer. Make sure you’ve gone through [part 4](/tutorial/part-four/), [part 5](/tutorial/part-five/), and [part 6](/tutorial/part-six/) before continuing here.
+> Ce tutoriel est inclu dans la série sur les niveau de données de Gatsby. Soyez sur d'avoir lu [part 4](/tutorial/part-four/), [part 5](/tutorial/part-five/), et [part 6](/tutorial/part-six/) avant de continuer ici.
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:7]:
```suggestion
> Ce tutoriel est inclu dans la série sur les couches de données de Gatsby. Soyez sur d'avoir lu [la partie 4](/tutorial/part-four/), [la partie 5](/tutorial/part-five/), et [la partie 6](/tutorial/part-six/) avant de continuer ici.
```

File: docs/tutorial/part-seven/index.md [hunk 2]
```diff
@@ -1,61 +1,54 @@
...
 
-## What's in this tutorial?
+## Qu'y a-t-il dans ce tutoriel?
 
-In the previous tutorial, you created a nice index page that queries markdown
-files and produces a list of blog post titles and excerpts. But you don't want to just see excerpts, you want actual pages for your
-markdown files.
+Dans le tutoriel précédent vous avez créé une belle page d'accueil qui va chercher des fichiers
+markdown and produit une liste d'article de blog avec titre et extraits. Mais vous ne voulez pas juste voir
+les extraits mais les pages complètes pour les fichiers markdown.
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:13]:
```suggestion
les extraits mais des pages complètes pour chacun de vos fichiers markdown.
```

File: docs/tutorial/part-seven/index.md [hunk 3]
```diff
@@ -1,61 +1,54 @@
...
 
-You could continue to create pages by placing React components in `src/pages`. However, you'll
-now learn how to _programmatically_ create pages from _data_. Gatsby is _not_
-limited to making pages from files like many static site generators. Gatsby lets
-you use GraphQL to query your _data_ and _map_ the query results to _pages_—all at build
-time. This is a really powerful idea. You'll be exploring its implications and
-ways to use it for the remainder of this part of the tutorial.
+Vous pouvez continuer à créer des pages en plaçant des composants React dans `src/pages`. Par contre vous
+allez maintenant apprendre comment _programmatiquement_ créer des pages depuis _data_. Gatsby _n'est_ pas
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:16]:
```suggestion
allez maintenant apprendre comment _programmatiquement_ créer des pages à partir de _données_. Gatsby _n'est_ pas
```

File: docs/tutorial/part-seven/index.md [hunk 4]
```diff
@@ -1,61 +1,54 @@
...
+limité à concevoir des pages depuis des fichiers comme beaucoup de générateurs de sites statiques. Gatsby vous permet d'utiliser GraphQL pour rechercher vos _données_ et _mapper_ le résultat en _pages_-tout cela à l'état de build. C'est réellement une idée puissante. Vous allez explorer ces implications et les façons de s'en servir sur la suite de ce tutoriel.
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:17]:
```suggestion
limité à concevoir des pages depuis des fichiers comme beaucoup de générateurs de sites statiques. Gatsby vous permet d'utiliser GraphQL pour rechercher vos _données_ et _transformer_ le résultat en _pages_ - tout cela au moment du build. C'est réellement une idée puissante. Vous allez explorer ces implications et les façons de s'en servir sur la suite de ce tutoriel.
```

Comment (mindsers) [reply docs/tutorial/part-seven/index.md:17]: I'm not comfortable with "rechercher vos données". It's like you don't find you data and gatsby give you tools to investigate.

I don't know how to say it better though

File: docs/tutorial/part-seven/index.md [hunk 5]
```diff
@@ -1,61 +1,54 @@
...
 
-Let's get started.
+Commencons
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:19]:
```suggestion
Commençons.
```

File: docs/tutorial/part-seven/index.md [hunk 6]
```diff
@@ -1,61 +1,54 @@
...
 
-## Creating slugs for pages
+## Créer des slugs pour les pages
 
-Creating new pages has two steps:
+Créer de nouvelles pages nécessite deux étapes:
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:23]: There is a space before colon in French.
```suggestion
Créer de nouvelles pages nécessite deux étapes :
```

File: docs/tutorial/part-seven/index.md [hunk 7]
```diff
@@ -1,61 +1,54 @@
...
+_**Note**: Souvent les données sources vont directement fournir un slug ou chemin pour le contenu - quand vous travaillez avec un tel système (ex. un CMS), vous n'avez pas besoin de créer le slug vous même comme vous le faites avec les fichiers markdown._
 
-To create your markdown pages, you'll learn to use two Gatsby APIs:
-[`onCreateNode`](/docs/node-apis/#onCreateNode) and
-[`createPages`](/docs/node-apis/#createPages). These are two workhorse APIs
-you'll see used in many sites and plugins.
+Pour créer vos pages markdown vous allez apprendre à utiliser deux APIs de Gatsby:
+[`onCreateNode`](/docs/node-apis/#onCreateNode)
+[`createPages`](/docs/node-apis/#createPages).
+Ces deux APIs sont utilisées dans de nombreux sites et plugins
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:33]:
```suggestion
Ces deux APIs sont utilisées dans de nombreux sites et plugins.
```

File: docs/tutorial/part-seven/index.md [hunk 8]
```diff
@@ -1,61 +1,54 @@
...
+2.  Créer la page.
 
-_**Note**: Often data sources will directly provide a slug or pathname for content — when working with one of those systems (e.g. a CMS), you don't need to create the slugs yourself as you do with markdown files._
+_**Note**: Souvent les données sources vont directement fournir un slug ou chemin pour le contenu - quand vous travaillez avec un tel système (ex. un CMS), vous n'avez pas besoin de créer le slug vous même comme vous le faites avec les fichiers markdown._
 
-To create your markdown pages, you'll learn to use two Gatsby APIs:
-[`onCreateNode`](/docs/node-apis/#onCreateNode) and
-[`createPages`](/docs/node-apis/#createPages). These are two workhorse APIs
-you'll see used in many sites and plugins.
+Pour créer vos pages markdown vous allez apprendre à utiliser deux APIs de Gatsby:
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:30]:
```suggestion
Pour créer vos pages markdown vous allez apprendre à utiliser deux APIs de Gatsby :
```

File: docs/tutorial/part-seven/index.md [hunk 9]
```diff
@@ -1,61 +1,54 @@
...
+[`onCreateNode`](/docs/node-apis/#onCreateNode)
+[`createPages`](/docs/node-apis/#createPages).
+Ces deux APIs sont utilisées dans de nombreux sites et plugins
 
-We do our best to make Gatsby APIs simple to implement. To implement an API, you export a function
-with the name of the API from `gatsby-node.js`.
+Nous nous efforçons de faire les APIs de Gatsby simple à implementer. Pour implémenter une API vous devez exporter la fonction avec le nom de l'API depuis `gatsby-node.js`.
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:35]:
```suggestion
Nous nous efforçons de rendre les APIs de Gatsby simple à implementer. Pour implémenter une API vous devez exporter une fonction avec le nom de l'API depuis `gatsby-node.js`.
```

File: docs/tutorial/part-seven/index.md [hunk 10]
```diff
@@ -1,61 +1,54 @@
...
-Stop and restart the development server. As you do, you'll see quite a few newly
-created nodes get logged to the terminal console.
+Eteignez et relancez le serveur de développement. En le faisant, vous allez voir de nouveaux noeuds créés être affiché dans la console du terminale
 
-Use this API to add the slugs for your markdown pages to `MarkdownRemark`
-nodes.
+Utilisez cette API pour ajouter le slug pour les pages markdown dans le nodes `MarkdownRemark`.
 
-Change your function so it now only logs `MarkdownRemark` nodes.
+Changez la fonction pour qu'elle affiche  maintenant seulement les noeuds `MarkdownRemark`.
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:51]:
```suggestion
Changez la fonction pour qu'elle affiche maintenant seulement les nœuds `MarkdownRemark`.
```

File: docs/tutorial/part-seven/index.md [hunk 11]
```diff
@@ -1,61 +1,54 @@
...
   console.log(node.internal.type)
 }
 ```
 
-This `onCreateNode` function will be called by Gatsby whenever a new node is created (or updated).
+Cette fonction `onCreateNode` va être appelée par Gatsby a chaque fois qu'un nouveau noeud est créé (ou mis à jour).
 
-Stop and restart the development server. As you do, you'll see quite a few newly
-created nodes get logged to the terminal console.
+Eteignez et relancez le serveur de développement. En le faisant, vous allez voir de nouveaux noeuds créés être affiché dans la console du terminale
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:47]:
```suggestion
Eteignez et relancez le serveur de développement. En le faisant, vous allez voir de nouveaux nœuds créés être affiché dans la console du terminale
```

File: docs/tutorial/part-seven/index.md [hunk 12]
```diff
@@ -67,11 +60,7 @@ exports.onCreateNode = ({ node }) => {
 }
 ```
 
-You want to use each markdown file name to create the page slug. So
-`pandas-and-bananas.md` will become `/pandas-and-bananas/`. But how do you get
-the file name from the `MarkdownRemark` node? To get it, you need to _traverse_
-the "node graph" to its _parent_ `File` node, as `File` nodes contain data you
-need about files on disk. To do that, modify your function again:
+Vous voulez utiliser chaque nom de fichier markdown pour créer le slug de la page. Donc `pandas-et-bananes.md` va devenir `/pandas-et-bananes/`. Mais comment obtenir le nom du fichier depuis le noeud `MarkdownRemark`? POur y accéder vous devez _traverser_ le "graph de noeud" à son noeud _parent_ `File`, ce noeud contenant la donnée recherchée au sujet des fichiers sur le disque. Pour cela, modifiez votre fonction encore une fois:
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:63]:
```suggestion
Vous voulez utiliser chaque nom de fichier markdown pour créer le slug de la page. Donc `pandas-et-bananes.md` va devenir `/pandas-et-bananes/`. Mais comment obtenir le nom du fichier depuis le nœud `MarkdownRemark` ? Pour y accéder vous devez _traverser_ le "graph de nœud" jusqu'à son nœud _parent_ `File`, puisque les nœuds `File` contiennent les données dont vous avez besoin sur les fichiers sur le disque. Pour cela, modifiez votre fonction encore une fois :
```

File: docs/tutorial/part-seven/index.md [hunk 13]
```diff
@@ -85,14 +74,11 @@ exports.onCreateNode = ({ node, getNode }) => {
 }
 ```
 
-After restarting your development server, you should see the relative paths for your two markdown
-files print to the terminal screen.
+Après avoir relancé votre serveur de développement vous devriez voir affiche sur l'écran du terminal les chemins relatifs pour vos deux fichiers markdown
```
Comment (mindsers) [inline docs/tutorial/part-seven/index.md:77]:
```suggestion
Après avoir relancé votre serveur de développement vous devriez voir sur l'écran du terminal les chemins relatifs pour vos deux fichiers markdown.
```

**Review by mindsers — CHANGES_REQUESTED** [review]
Thanks for you PR ❤️ 

This is not a full review but we can go step by step

**Comment by dantereve** [comment]
As we still have l no news from you I'm closing this PR so that someone can take over your translation and update it with recent changes in the main repo. As it's one of your private branches we can't modify this ourselves. Please feel free to take another page or even update this one 👌

**Pull request closed without merging**